design/KruizeLocalAPI.md

```diff
@@ -0,0 +1,1986 @@
+                    "module-configmap-reloader": {
+                      "container_name": "module-configmap-reloader",
+                      "container_image_name": "jimmidyson/configmap-reload:v0.5.0"
+                    },
+                    "blackbox-exporter": {
+                      "container_name": "blackbox-exporter",
+                      "container_image_name": "quay.io/prometheus/blackbox-exporter:v0.18.0"
+                    }
+                  }
+                },
+                "prometheus-operator": {
+                  "workload_name": "prometheus-operator",
+                  "workload_type": "deployment",
+                  "containers": {
+                    "kube-rbac-proxy": {
+                      "container_name": "kube-rbac-proxy",
+                      "container_image_name": "quay.io/brancz/kube-rbac-proxy:v0.8.0"
+                    },
+                    "prometheus-operator": {
+                      "container_name": "prometheus-operator",
+                      "container_image_name": "quay.io/prometheus-operator/prometheus-operator:v0.47.0"
+                    }
+                  }
+                }
+              }
+            },
+            "kube-public": {
+              "namespace": "kube-public"
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+</details>
+
+<br>
+
+**Request with datasource name and cluster name parameter**
+
+`GET /dsmetadata`
+
+`curl -H 'Accept: application/json' "http://<URL>:<PORT>/dsmetadata?datasource=<datasource_name>&cluster_name=<cluster_name>"`
+
+Returns the list of namespaces present in the specified cluster name and datasource
+
+**Response for datasource name - `prometheus-1` and cluster name - `default`**
+
+<details>
+<summary><b>Example Response</b></summary>
+
+### Example Response
+
+```json
+{
+  "datasources": {
+    "prometheus-1": {
+      "datasource_name": "prometheus-1",
+      "clusters": {
+        "default": {
+          "cluster_name": "default",
+          "namespaces": {
+            "default": {
+              "namespace": "default"
+            },
+            "cadvisor": {
+              "namespace": "cadvisor"
+            },
+            "kube-node-lease": {
+              "namespace": "kube-node-lease"
+            },
+            "kube-system": {
+              "namespace": "kube-system"
+            },
+            "monitoring": {
+              "namespace": "monitoring"
+            },
+            "kube-public": {
+              "namespace": "kube-public"
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+</details>
+
+<br>
+
+**Request with datasource name, cluster name and verbose parameters**
+
+`GET /dsmetadata`
+
+`curl -H 'Accept: application/json' "http://<URL>:<PORT>/dsmetadata?datasource=<datasource_name>&cluster_name=<cluster_name>&verbose=true"`
+
+Returns the container-level metadata of all the namespaces present in the specified cluster name and datasource
```

Comment:
Please mention the difference with and without verbose more clearly

Reply:
Updated description for verbose parameter